macaroon.js

```diff
@@ -3,9 +3,30 @@
 const sjcl = require('sjcl');
 const nacl = require('tweetnacl');
 nacl.util = require('tweetnacl-util');
+const varint = require('varint');
 
 const NONCELEN = 24;
 
+const V2_TYPES = {
+  LOCATION: 1,
+  IDENTIFIER: 2,
+  VID: 4,
+  SIGNATURE: 6
+};
+
+// Uses protobuf-style varints
+function toVarintBuf(number) {
+  return Buffer.from(varint.encode(number));
```

Comment:
If you use something like browserify or webpack it would include a shim for the Buffer module. If you want to make the module work without one of those build tools you could make it depend directly on https://github.com/feross/buffer

Reply:
I didn't want the API to expose Buffer directly,   thus forcing users to use something that doesn't feel native to the platform. Uint8Array seems like nice shared functionality that does the job we need.